src/RulesFactory.js

```diff
@@ -0,0 +1,69 @@
+import warning from 'warning'
+
+import RegularRule from './rules/RegularRule'
+import SimpleRule from './rules/SimpleRule'
+import KeyframeRule from './rules/KeyframeRule'
+import ConditionalRule from './rules/ConditionalRule'
+import FontFaceRule from './rules/FontFaceRule'
+
+export default class RulesFactory {
+  constructor() {
+    this.classes = {
+      '@charset': SimpleRule,
+      '@import': SimpleRule,
+      '@namespace': SimpleRule,
+      '@keyframes': KeyframeRule,
+      '@media': ConditionalRule,
+      '@supports': ConditionalRule,
+      '@font-face': FontFaceRule
+    }
+    this.regExp = this.buildRegExp()
```

Comment:
Why use a constructor for these use cases? Stylistic preference? I noticed that we have `stage-0` enabled, and class properties are already `stage-2`:

```js
class RulesFactory {
  classes = {
    '@charset': SimpleRule,
  }

  regExp = this.buildRegExp()
}
```